syntax = "proto3";
package alert;




import "monitor_sdk/model/monitor/alert_event.proto";

/*
GetAlertEventDetail请求
*/
message GetAlertEventDetailRequest {
    /*
    告警 ID
    */
    string _id = 1 ;
    /*
    开始时间, e.g.: -1h 过去1小时, -1d 过去1天, -30m 过去30分钟, 1490967693 unix 时间戳
    */
    string st = 2 ;
    /*
    结束时间, e.g.: -1h 过去1小时, -1d 过去1天, -30m 过去30分钟, 1490967693 unix 时间戳。默认为当前时间
    */
    string et = 3 ;
}

/*
GetAlertEventDetail返回
*/
message GetAlertEventDetailResponse {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    消息
    */
    string msg = 2 ;
    /*
    页码
    */
    int32 page = 3 ;
    /*
    分页大小
    */
    int32 page_size = 4 ;
    /*
    总数
    */
    int32 total = 5 ;
    /*
    告警事件列表，返回的是一个列表，只有一个告警事件的列表当作详情接口的返回
    */
    repeated monitor.AlertEvent data = 6 ;
}

/*
GetAlertEventDetailApi返回
*/
message GetAlertEventDetailResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    alert.GetAlertEventDetailResponse data = 4 ;
}

